view/frontend/web/js/view/payment/method-renderer/omise-offsite-internetbanking-method.js

```diff
@@ -82,54 +111,58 @@ define(
+                Omise.setPublicKey(this.getPublicKey());
+                Omise.createSource(this.omiseOffsite(), { amount: this.getOrderAmount(), currency: this.getOrderCurrency(), }, function (statusCode, response) {
+                    self.omiseSourceToken(response.id);
+
+                    self.getPlaceOrderDeferredObject()
+                        .fail(
+                            function (response) {
+                                errorProcessor.process(response, self.messageContainer);
+                                fullScreenLoader.stopLoader();
+                                self.isPlaceOrderActionAllowed(true);
+                            }
+                        ).done(
+                            function (response) {
+                                var self = this;
+
+                                var serviceUrl = urlBuilder.createUrl(
+                                    '/orders/:order_id/omise-offsite',
+                                    {
+                                        order_id: response
                                     }
-                                )
-                                .done(
-                                    function (response) {
-                                        if (response) {
-                                            $.mage.redirect(response.authorize_uri);
-                                        } else {
+                                );
+
+                                storage.get(serviceUrl, false)
+                                    .fail(
+                                        function (response) {
                                             errorProcessor.process(response, self.messageContainer);
                                             fullScreenLoader.stopLoader();
                                             self.isPlaceOrderActionAllowed(true);
                                         }
-                                    }
-                                );
-                        }
-                    );
-
-                return true;
+                                    )
+                                    .done(
+                                        function (response) {
+                                            if (response) {
+                                                $.mage.redirect(response.authorize_uri);
+                                            } else {
+                                                errorProcessor.process(response, self.messageContainer);
+                                                fullScreenLoader.stopLoader();
+                                                self.isPlaceOrderActionAllowed(true);
+                                            }
+                                        }
+                                    );
+                            }
+                        );
+                    return true;
+                });
             }
         });
     }
```

Comment:
A lot of the code across the two main `js` files for the new payment methods seems repeated. I wonder if there maybe some scope to refactor into something more generic? Would certainly help us if another offline payment option comes along

Reply:
i will think how to integrate it during our 2 weeks... after current sprint